pyodide_build/pypabuild.py

```diff
@@ -191,6 +209,45 @@ def _build_in_isolated_env(
             )
 
 
+def _build_in_current_env(
+    build_env: Mapping[str, str],
+    srcdir: Path,
+    outdir: str,
+    distribution: Literal["sdist", "wheel"],
+    config_settings: ConfigSettingsType,
+    skip_dependency_check: bool = False,
+) -> str:
+    with common.replace_env(build_env):
+        # Setup sysconfigdata path in environment
+        sysconfigdata_name = get_build_flag("SYSCONFIG_NAME")
+        sysconfig_dir = Path(get_build_flag("TARGETINSTALLDIR")) / "sysconfigdata"
+        env = os.environ.copy()
+        env["_PYTHON_SYSCONFIGDATA_NAME"] = sysconfigdata_name
+        env["PYTHONPATH"] = f"{str(sysconfig_dir)}:{env.get('PYTHONPATH', '')}"
+
+        with common.replace_env(env):
+            builder = _ProjectBuilder(srcdir, runner=_gen_runner(build_env))
+
+            if not skip_dependency_check:
+                missing = builder.check_dependencies(
```

Comment:
👍 

[`check_dependencies`](https://github.com/pypa/build/blob/811c9db0dcac14b7b32f08ee57d09af3666979d4/src/build/_builder.py#L242) looks like a union of

1. get_requires_for_build (== dependencies for the build backend such as setuptools, meson)
2. build_system_requires (== dependencies in the pyproject.toml)

In the isolated build, we've been [dealing with them separately](https://github.com/pyodide/pyodide-build/blob/71aa0b5ae55b196c4b3444434c67a9541dd382c9/pyodide_build/pypabuild.py#L160-L167). I guess we can use `check_dependencies` there too (no action required for this PR).

Reply:
Moved this to a new issue: #175.